ext/nokogiri/nokogiri.c

```diff
@@ -52,27 +52,22 @@ void noko_init_test_global_handlers(void);
 static ID id_read, id_write, id_external_encoding;
 
 
-#ifndef HAVE_VASPRINTF
 /*
  * Thank you Geoffroy Couprie for this implementation of vasprintf!
  */
 int
-vasprintf(char **strp, const char *fmt, va_list ap)
+noko_vasprintf(char **strp, const char *fmt, va_list ap)
 {
   /* Mingw32/64 have a broken vsnprintf implementation that fails when
    * using a zero-byte limit in order to retrieve the required size for malloc.
    * So we use a one byte buffer instead.
    */
   char tmp[1];
   int len = vsnprintf(tmp, 1, fmt, ap) + 1;
-  char *res = (char *)malloc((unsigned int)len);
-  if (res == NULL) {
-    return -1;
-  }
+  char *res = (char *)ruby_xmalloc((unsigned int)len);
```

Comment:
`ruby_xmalloc()` returns `void *`, the return values from the `malloc` family of functions should not be cast.

With the previous change above, this line can become
```c
char *rest = ruby_xmalloc(len);
```